framework/set/setETCDSnapshots.go

```diff
@@ -0,0 +1,38 @@
+package framework
+
+import (
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/rancher/tfp-automation/config"
+	"github.com/zclconf/go-cty/cty"
+)
+
+// setCreateRKE2K3SSnapshot is a function that will create a snapshot for a RKE2/K3S cluster.
+func setCreateRKE2K3SSnapshot(terraformConfig *config.TerraformConfig, rkeConfigBlockBody *hclwrite.Body) {
+	createSnapshotBlock := rkeConfigBlockBody.AppendNewBlock("etcd_snapshot_create", nil)
+	createSnapshotBlockBody := createSnapshotBlock.Body()
+
+	generation := int64(1)
+
+	if createSnapshotBlockBody.GetAttribute("generation") == nil {
+		createSnapshotBlockBody.SetAttributeValue("generation", cty.NumberIntVal(generation))
+	} else {
+		createSnapshotBlockBody.SetAttributeValue("generation", cty.NumberIntVal(generation+1))
+	}
+}
+
+// setRestoreRKE2K3SSnapshot is a function that will restore a snapshot for a RKE2/K3S cluster.
```

Comment:
can you modify this comment to specify that it is being set in main.tf, not actually restoring the resource in this function?

Reply:
Resolved